Data/ByteString.hs

```diff
@@ -1419,18 +1422,21 @@ findSubstring pat src
 findSubstrings :: ByteString -- ^ String to search for.
                -> ByteString -- ^ String to seach in.
                -> [Int]
-findSubstrings pat src
-    | null pat        = [0 .. ls]
-    | otherwise       = search 0
+findSubstrings pat = go
   where
-    lp = length pat
-    ls = length src
-    search !n
-        | (n > ls - lp) || null b = []
-        | otherwise = let k = n + length a
-                      in  k : search (k + lp)
+    breakPat = breakSubstring pat
```

Comment:
Again, we should comment here why we do it like this, since it's important and not immediately obvious from this definition (though it is of course if you look at the docs for `breakSubstring`).